packages/aws-cdk-lib/aws-stepfunctions/lib/activity.ts

```diff
@@ -57,14 +79,23 @@ export class Activity extends Resource implements IActivity {
    */
   public readonly activityName: string;
 
+  /**
+   * Default value for `kmsDataKeyReusePeriodSeconds`
+   * @see https://docs.aws.amazon.com/step-functions/latest/dg/encryption-at-rest.html#cfn-resources-for-encryption-configuration
+   */
+  private readonly defaultPeriodSeconds = 300;
+
   constructor(scope: Construct, id: string, props: ActivityProps = {}) {
     super(scope, id, {
       physicalName: props.activityName ||
-                Lazy.string({ produce: () => this.generateName() }),
+        Lazy.string({ produce: () => this.generateName() }),
     });
 
+    validateEncryptionConfiguration(props.kmsKey, props.kmsDataKeyReusePeriodSeconds);
+
     const resource = new CfnActivity(this, 'Resource', {
       name: this.physicalName!, // not null because of above call to `super`
+      encryptionConfiguration: constructEncryptionConfiguration(props, this.defaultPeriodSeconds),
```

Comment:
You scoped down the encryption context for StateMachine but not Activity. Why?

https://docs.aws.amazon.com/step-functions/latest/dg/encryption-at-rest.html#scope-down-kms-permission-policies-with-conditions

Reply:
That was a miss, will add that, thanks.